src/compile.ts

```diff
@@ -269,6 +287,54 @@ export function compile(
         );
       }
 
+      function isNewFunctionlessFunction(
+        node: ts.Node
+      ): node is FunctionInterface {
+        return (
+          ts.isNewExpression(node) &&
+          isFunctionlessClassOfKind(
+            node.expression,
+            Function.FunctionlessType
+          ) &&
+          // only take the form with the arrow function at the end.
+          (node.arguments?.length === 3 || node.arguments?.length === 4) &&
+          ts.isArrowFunction(node.arguments[2])
+        );
+      }
+
+      /**
+       * Heuristically evaluate the fqn of a symbol to be in a module and of a type name.
+       *
+       * /somePath/node_modules/{module}/somePath.{typeName}
+       *
+       * isInstanceOf(typeSymbol, "constructs", "Construct")
+       * ex: /home/sussmans/functionless/node_modules/constructs/lib/index.js
+       */
+      function isInstanceOf(
+        symbol: ts.Symbol,
+        module: string,
+        typeName: string
+      ) {
+        const find = /.*\/node_modules\/([^\/]*)\/.*\.(.*)$/g.exec(
+          checker.getFullyQualifiedName(symbol)
+        );
+
+        const [_, mod, type] = find ?? [];
+
+        return mod === module && type === typeName;
+      }
+
+      function isCDKConstruct(type: ts.Type): boolean {
+        const typeSymbol = type.getSymbol();
+
+        return (
+          ((typeSymbol &&
+            isInstanceOf(typeSymbol, "constructs", "Construct")) ||
+            type.getBaseTypes()?.some((t) => isCDKConstruct(t))) ??
+          false
+        );
+      }
```

Comment:
I would check for the node property and then check for many of the properties on the node class.

Reply:
Answered here: https://github.com/functionless/functionless/pull/103#discussion_r884063485

I think this will work. The chance that a package name and type name collide in the same code base is low.